src/main.rs

```diff
@@ -15,40 +15,30 @@ extern crate structopt;
 
 use cargo::util::CliError;
 use cargo_ebuild::run;
+use cargo_ebuild::Command;
 use std::process;
-use structopt::clap::AppSettings;
 use structopt::StructOpt;
+use structopt::clap::AppSettings;
 
 #[derive(StructOpt, Debug)]
-struct Args {
+#[structopt(name = "cargo", raw(setting = "AppSettings::ColoredHelp"))]
+struct Opt {
     /// Silence all output
     #[structopt(short = "q", long = "quiet")]
     quiet: bool,
     /// Verbose mode (-v, -vv, -vvv, etc.)
     #[structopt(short = "v", long = "verbose", parse(from_occurrences))]
     verbose: usize,
-}
-
-#[derive(StructOpt, Debug)]
-#[structopt(bin_name = "cargo")]
-enum Opt {
-    #[structopt(
-        name = "ebuild",
-        raw(
-            setting = "AppSettings::UnifiedHelpMessage",
-            setting = "AppSettings::DeriveDisplayOrder",
-            setting = "AppSettings::DontCollapseArgsInUsage"
-        )
-    )]
     /// Generates an ebuild for a given Cargo project
-    Ebuild(Args),
+    #[structopt(subcommand)]
+    cmd: Option<Command>,
 }
 
 fn main() {
-    let Opt::Ebuild(opt) = Opt::from_args();
+    let opt = Opt::from_args();
 
     // run the actual code
-    if let Err(e) = run(opt.verbose as u32, opt.quiet) {
+    if let Err(e) = run(opt.cmd, opt.verbose as u32, opt.quiet) {
```

Comment:
This commit is mostly centered around this change. But then its removed in the next commit.

Reply:
here verbosity and quiet are used in the library, in future changes not.